meta-ostro/classes/image-dsk.bbclass

```diff
@@ -24,16 +24,104 @@
 #   This is required to identify it and pass its Partition UUID to the kernel, for booting.
 
 
-COMPRESSIONTYPES_append = " vdi"
+COMPRESSIONTYPES_append = " vdi ova"
 COMPRESS_CMD_vdi = "qemu-img convert -O vdi ${IMAGE_NAME}${IMAGE_NAME_SUFFIX}.${type} ${IMAGE_NAME}${IMAGE_NAME_SUFFIX}.${type}.vdi"
+
 COMPRESS_DEPENDS_vdi = "qemu-native"
+COMPRESS_DEPENDS_ova = "virtualbox-native"
 
 IMAGE_DSK_ACTIVE = "${@ bool([x for x in d.getVar('IMAGE_FSTYPES', True).split() if x == 'dsk' or x.startswith('dsk.')])}"
+
+
 python () {
     if d.getVar('IMAGE_DSK_ACTIVE', True) == 'True':
         d.setVar('IMAGE_NAME_SUFFIX', '')
 }
 
+
+create_ova() {
+
+
+  if [ "x${MACHINE}x" = "xintel-corei7-64x" -o "x${MACHINE}x" = "xintel-quarkx" ]; then
```

Comment:
Let's start with intel-corei7-64 only. The if [] can exit if not intel-corei7-64.